ml_model/model/feature_selection/feature_selection.py

```diff
@@ -0,0 +1,114 @@
+
+import csv
+
+from sklearn.linear_model import LogisticRegression
+
+df = pd.read_csv('ml_model/src/main/model/data/understandability_with_warnings.csv')
+categorical_features = ["Cyclomatic complexity", "IMSQ (min)","MIDQ (min)","AEDQ (min)","EAP (min)","IMSQ (avg)", "MIDQ (avg)","AEDQ (avg)","EAP (avg)","IMSQ (max)","MIDQ (max)","AEDQ (max)", "EAP (max)","Readability", "ITID (avg)","NMI (avg)","NMI (max)","CIC (avg)","CIC (max)", "CICsyn (avg)", "CICsyn (max)", "CR", "NM (avg)", "NM (max)", "TC (avg)","TC (min)","TC (max)","#assignments (avg)","#blank lines (avg)","#commas (avg)","#comments (avg)","#comparisons (avg)", "Identifiers length (avg)", "#conditionals (avg)","Indentation length (avg)","#keywords (avg)","Line length (avg)", "#loops (avg)", "#identifiers (avg)", "#numbers (avg)","#operators (avg)", "#parenthesis (avg)", "#periods (avg)","#spaces (avg)","Identifiers length (max)","Indentation length (max)", "#keywords (max)", "Line length (max)",	"#identifiers (max)","#numbers (max)","#characters (max)","#words (max)","Entropy","Volume","LOC","#assignments (dft)","#commas (dft)","#comments (dft)", "#comparisons (dft)","#conditionals (dft)","Indentation length (dft)","#keywords (dft)","Line length (dft)","#loops (dft)","#identifiers (dft)","#numbers (dft)","#operators (dft)","#parenthesis (dft)","#periods (dft)","#spaces (dft)","Comments (Visual X)","Comments (Visual Y)","Identifiers (Visual X)","Identifiers (Visual Y)","Keywords (Visual X)","Keywords (Visual Y)","Numbers (Visual X)","Numbers (Visual Y)","Strings (Visual X)","Strings (Visual Y)","Literals (Visual X)","Literals (Visual Y)","Operators (Visual X)","Operators (Visual Y)","Comments (area)","Identifiers (area)","Keywords (area)","Numbers (area)","Strings (area)","Literals (area)","Operators (area)",	"Keywords/identifiers (area)","Numbers/identifiers (area)","Strings/identifiers (area)","Literals/identifiers (area)","Operators/literals (area)","Numbers/keywords (area)","Strings/keywords (area)","Literals/keywords (area)","Operators/keywords (area)", "#aligned blocks","Extent of aligned blocks","#nested blocks (avg)","#parameters","#statements", "warnings_checker_framework", "warnings_typestate_checker", "warnings_infer"	, "warnings_openjml",	"warning_sum"]
+categorical_target = ["PBU", "BD50", "ABU50"]
+X = df[categorical_features]
+y = df[categorical_target]
+
+stat_header = {'input feature':'', 'output_feature': '', 'tau': 0, 'p-value':0}
+
+
+######################################
+## Classification feature selection ##
+######################################
+## Numeric input, categorical output
+
+## FILTER METHODS ##
+## write header
+with open('ml_model/src/main/model/feature_selection/kendall_features.csv', "w") as csv_file:
+    writer = csv.DictWriter(csv_file, fieldnames=stat_header.keys())
+    writer.writeheader()
+## Kendals Tau
+for feature in categorical_features:
+    for target in categorical_target:
+        tau,p = kendalltau(X[feature], y[target])
```

Comment:
Let's double check how NAs are treated

Reply:
kendalltau(X[feature1], X[feature2], nan_policy='omit') - This will ensure to perform pair-wise comparisons where both the data points have values (i.e not NAN)